jujugui/static/gui/src/app/app.js

```diff
@@ -2585,6 +2591,18 @@ YUI.add('juju-gui', function(Y) {
           );
         }, this);
       }
+      // Retrieve maas credentials should they exist.
+      // NB: this assumes that a maas cloud cannot be added to a multi-cloud
+      // controller. If this changes in the future, this code will need to be
+      // called somewhere else. - Makyo 2017-02-16
+      this.controllerAPI.getDefaultCloudName((error, name) => {
+        this.controllerAPI.getClouds([name], (error, data) => {
+          if (data['.names']) {
```

Comment:
We should check the error here as well, for the second call, maybe
```javascript
this.controllerAPI.getDefaultCloudName((error, name) => {
  if (error) {
    console.error('cannot retrieve cloud name for maas: ' + error);
    return;
  }
  this.controllerAPI.getClouds([name], (error, clouds) => {
    const err = error || clouds[name].err;
    if (err) {
      console.error('cannot retrieve clouds for maas: ' + err);
      return;
    }
    this.controllerAPI.set('maasServer', clouds[name].endpoint);
  });
});
```
What do you think?